packages/javascript-sdk/src/logger/index.ts

```diff
@@ -0,0 +1,47 @@
+import Config from '../config/index';
+
+interface LoggerFunctions {
+  warnFn?: (...msgs: unknown[]) => void;
+  errorFn?: (...msgs: unknown[]) => void;
+  logFn?: (...msgs: unknown[]) => void;
+}
+
+function customLogger() {
+  let frWarn = console.log,
+    frError = console.error,
+    frLog = console.log;
+
+  return {
+    addLogger({
+      logFn = console.log,
+      warnFn = console.warn,
+      errorFn = console.error,
+    }: LoggerFunctions) {
+      frLog = logFn;
+      frWarn = warnFn;
+      frError = errorFn;
+    },
+    shouldLog() {
```

Comment:
Not a huge fan of multi-word properties, if we can avoid them. Would `enabled` work as the method name? What's more, I'm not normally a fan of using "[getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get)" and "setters", but I feel it might be fairly useful here. That way, we can also avoid making the dev call the function. They could just `frLogger.enabled` and get back the boolean?

Reply:
We can move this into the closure if thats your suggestion, thats easy enough

Reply:
I'm not sure if you want them to be able to use the logger to verify if logging is enabled?

Right now if someone wants to check on Logging, they can do so through `Config.logLevel` but we could move all of this to the Logger and even remove `LogLevel` from `Config`, is that your thought?